### DISCUSSION TITLE
High cost of clone cache in CI

### DISCUSSION THREAD

**Discussion by jgsuess** [oracle/macaron#338] [Q&A]
According to the documentation, substantial amounts of data are cloned into the container. This means high transfer cost and lots of runtime. In CI runs, these clones should hence be stored in caches for the runs and just updated. Where are the corresponding directories located? Are the repository clones somehow tainted by the parameters or can one cache be used for all invocations?

**Comment by behnazh-w** [comment]
Currently we don't cache the cloned repositories to make sure CI runs do not affect each other, which has the downside of longer CI execution time.

**Comment by jgsuess** [comment]
Cloning alone should not have an effect on the run, unless someone does something really vile and publishes divergent changes. It is the state of the work area that should be reset to the branch. The bulk of data is in the repo, which in git is add-only.
The only requirement on `macaron` that I can see would be to clone if the repo does not exist. Or am I missing something? The cost in execution for transferring whole repos would be *really significant*. The alternative would be shallow clones. But those can be tricky to handle.
